examples/go/fakemodem/modem.go

```diff
@@ -0,0 +1,271 @@
+/*
+ * Copyright 2019 Infostellar, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package main
+
+import (
+	"context"
+	"io/ioutil"
+	"log"
+	"sync"
+	"time"
+
+	"github.com/golang/protobuf/ptypes"
+
+	v1 "github.com/infostellarinc/go-stellarstation/api/v1"
+	api "github.com/infostellarinc/go-stellarstation/api/v1/groundstation"
+	radio "github.com/infostellarinc/go-stellarstation/api/v1/radio"
+)
+
+/*********
+ * Modem *
+ *********/
+
+// Modem does all of the actual work.
+type Modem struct {
+	config         *Config
+	groundstations map[string]GroundStationConfig      // Indexed by name
```

Comment:
I guess you only want to connect to one graoundstation and one satellite from the code at a time. If it is true, we may simplify the code by removing map.